index.ts

```diff
@@ -415,22 +415,23 @@ function genBech32(encoding: 'bech32' | 'bech32m') {
   const toWords = _words.encode;
   const fromWordsUnsafe = unsafeWrapper(fromWords);
 
-  function encode(
-    prefix: string,
+  function encode<Prefix extends string>(
+    prefix: Prefix,
     words: number[] | Uint8Array,
     limit: number | false = 90
-  ): string {
+  ): `${Lowercase<Prefix>}1${string}` {
     if (typeof prefix !== 'string')
       throw new Error(`bech32.encode prefix should be string, not ${typeof prefix}`);
     if (!Array.isArray(words) || (words.length && typeof words[0] !== 'number'))
       throw new Error(`bech32.encode words should be array of numbers, not ${typeof words}`);
     const actualLength = prefix.length + 7 + words.length;
     if (limit !== false && actualLength > limit)
       throw new TypeError(`Length ${actualLength} exceeds limit ${limit}`);
-    prefix = prefix.toLowerCase();
-    return `${prefix}1${BECH_ALPHABET.encode(words)}${bechChecksum(prefix, words, ENCODING_CONST)}`;
+    return `${prefix.toLowerCase()}1${BECH_ALPHABET.encode(words)}${bechChecksum(prefix, words, ENCODING_CONST)}` as `${Lowercase<Prefix>}1${string}`;
```

Comment:
why does bechChecksum receive non-lowercased prefix now?

Reply:
Whoops. It's because `prefix = prefix.toLowerCase()` was resulting in a type error so I had to change it, and I overlooked the second `prefix`.